### ISSUE TITLE
Release process documentation?

### ISSUE THREAD

**Issue by chrisballinger** [vitor251093/wineskin#23]
I've had trouble piecing together the various build products. Is this process documented anywhere? Ideally I want to automate nightly builds for Engines, Wrappers and Winery.

**Comment by Gcenx** [comment]
"Wineskin Winery.app" runs just fine after being built, only needs to be compressed if it's to be added to the server for downloading directly from within the app and then update the {server}/Wrapper/NewestVersion.txt
{server}/Wrapper/WineskinWinery.app.tar.7z
The above is basicly the same for everything else that needs to be hosted, engines text file just includes a list of available engines to download.


Wineskin.app needs to be placed inside.
>wineskin-{version master wrapper version number}.app
drive_c symlink to "Contents/Resources/drive_c"
Contents

Contents Folder>
>Frameworks
MacOS
Resources {included in the provided link)
Info.plist {included in the provided link} (includes the entries from my changes)

Frameworks Contents> 
>Frameworks>{needed dylibs} (I included current ones in the link provided)
Frameworks> {add ObjectiveC_Extension.framework}
Frameworks>WSX11Prefs.plist (included in the provided link)
Frameworks>bin (included in the provided link)

MacOS Contents>
>WineskinLancher {current version from project}
WineskinX11 (included in the provided link)

[Wineskin master wrapper template](https://mega.nz/#!zJYlTKIR!B8y31wT7phVbBq3XdpskV6xcl7t6bEYTXBIAaXi0jNs)

For engines it can be a pain since macports/homebrew do not download all the needed build dependancies.

I will link back here later I'm already working on an updated script that works from within "Wineskin Winery" for engine creation, so I'm sure you could get better idea from that

**Comment by chrisballinger** [comment]
Thanks! It would be great if we could figure out how to bundle the app in such a way where the "Wineskin Winery.app" bundle includes the latest "Wineskin.app" wrapper template, and at least one working Engine.

Also is there any way to run `Wineskin.app` or `WineskinLauncher` in the Xcode debugger? They normally want to live inside a wrapper.

**Comment by Gcenx** [comment]
I don't know if including it inside "Wineskin Winery.app" is a good idea.
That would mean needing to update "Wineskin Winery.app" a lot more to ensure future compatibility with macOS versions. But I can see the appeal "winebottler" did something along those lines.


Running "Wineskin.app" inside xcode debugger I have no clue how @vitor251093 might be better for answering that.

WineskinLauncher same thing, but I assume as long as you have the needed ObjectiveC extension I don't why it wouldn't be possible since you can pass commands directly to it. WSS- are the main functions.

**Comment by vitor251093** [comment]
> Thanks! It would be great if we could figure out how to bundle the app in such a way where the "Wineskin Winery.app" bundle includes the latest "Wineskin.app" wrapper template, and at least one working Engine.


> I don't know if including it inside "Wineskin Winery.app" is a good idea.
> That would mean needing to update "Wineskin Winery.app" a lot more to ensure future compatibility with macOS versions. But I can see the appeal "winebottler" did something along those lines.


I agree with @Gcenx . Keeping them separated makes we capable of releasing updates separately, and that's an advantage. While it would be interesting to release a package once in a while with an stable version of all of them, once the three of them get stable versions, they shouldn't be bonded together.

> Also is there any way to run Wineskin.app or WineskinLauncher in the Xcode debugger? They normally want to live inside a wrapper.

> Running "Wineskin.app" inside xcode debugger I have no clue how @vitor251093 might be better for answering that.


I usually build them and then in Debug mode, move to the insides of a wrapper and then, if I want logs, I run using the terminal. It certainly isn't the best method to do, but that's what I've done during my tests xD

Xcode can be configured to send the result of the build to a wrapper so you can run them, however it would require a wrapper with a specific name in a specific location. Since that changes from environment to environment, I decided to do this manual method instead of doing that.

Also, a reminder: while Wineskin.app can be build in Release mode, WineskinLauncher **can't**. The reason for that is simple: when an app is build for release today, it considers its Info.plist has unique, and gives the Info.plist an identifier of the binary. Summarising, if you run WineskinLauncher in Release mode, you can't change the Info.plist file of the wrapper, and that's a real problem for Wineskin wrappers.

**Comment by chrisballinger** [comment]
We can likely set up an Xcode build target that moves everything into the right place.

Even if we don't ship production builds this way, it would be great to remove manual steps from the debugging workflow.

**Comment by vitor251093** [comment]
Yeah, is the kind of thing that saves lots of time on term.

**Comment by chrisballinger** [comment]
On the road toward automating the builds, I made a tool to automate the creation of Engines: https://github.com/WineskinCommunity/WineskinEngineBuilder

**Comment by vitor251093** [comment]
Awesome! :D

**Comment by chrisballinger** [comment]
@Gcenx Haven't added a compile from source option yet, but have some of the scaffolding for it in there. Would you mind adding any documentation you have on building engines from scratch to this issue? https://github.com/WineskinCommunity/WineskinEngineBuilder/issues/1

**Comment by Gcenx** [comment]
@chrisballinger Sure I just need to dig it up from my backup, I could also 7zip the EngineBuild scripts I have along with everything building for 10.13.

Those are modified bash scripts based on what doh123 made but it works I built CX17.5.0 & CX17.5.0 64bit using them.

**Comment by niveuseverto** [comment]
@Gcenx can lend a hand here, if you restore Wineskin launcher template

**Comment by Gcenx** [comment]
> @Gcenx can lend a hand here, if you restore Wineskin launcher template

@niveuseverto if I have time tonight I can replace that with a newer mini dylib template.

If your only interested in building the project to use with official winehq wine builds then you can use the [prebuilt development](
https://github.com/Gcenx/WineskinServer) then you can just grab WineskinWinery.app.tar.7z from the Winery folder.

**Comment by niveuseverto** [comment]
I'm interested in this as a part of @foesmm to provide ability to create stable wineskins for supported gamebryo games, so building/packaging from source (including wine) is desirable

**Comment by FurloSK** [comment]
Hi @Gcenx , @vitor251093 , @niveuseverto – I know this is an old thread of an (mostly) abandoned project.
**Could any one of you please provide a new link to the "empty" Wrapper template?**

> Here is an updated [Wrapper template](https://mega.nz/#F!mMxRWIgY!20fphWENjCqpLiNOU04BWg) , I did not shrink it down to a mini dylib subset as you may need some of those so I left them all in. That template will allow the use of doh123 engines / Winehq engines / PortingKit engines. Copy the WineskinLauncher into "Contents/MacOS" and Wineskin.app into the same location as the "drive_c" symlink. Also add in the compiled "ObjectiveC_Extension.framework" into the templates "Frameworks" folder ~If you have any issues replace libfreetype.6.dylib from XQuartz 2.7.11~ Updated the template with libfreetype.6.dylib from XQuartz 2.7.11

The mega.nz links are dead and I am desperately trying to create the last "original" master wrapper from doh123 (v2.6.2), which is nowhere to be found (**for that matter, wouldn't you happen to have it?**). I was able to build it from the repository, but even with the instructions above from @Gcenx, I am still failing to create the wrapper app succesfully. I am creating a Wineskin archives repository here on GitHub where I would like to store all old Wineskin Wrappers, Engines and Wineries I could find on the internet, so that they are ready to use for Mojave+ users, since the [Kegworks-App/Kegworks](https://github.com/Kegworks-App/Kegworks) only works 10.15+.

Thank you very much!
